src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1219,9 +1219,34 @@ public async Task<(long ClusterEtag, string ClusterId)> GenerateClusterIdentityA
             {
                 [id] = identity
             };
-            (var index, var _) = await SendToLeaderAsync(new UpdateClusterIdentityCommand(databaseName, identities));
 
-            return index;
+            var (_, result) = await SendToLeaderAsync(new UpdateClusterIdentityCommand(databaseName, identities));
+
+            if (result == null)
+            {
+                throw new InvalidOperationException(
+                    $"Expected to get result from raft command that should update a cluster-wide identity, but didn't. Leader is {LeaderTag}, Current node tag is {NodeTag}.");
+            }
+
+            var resultDict = result as Dictionary<string, long> ?? throw new InvalidOperationException(
+                                 $"Expected to get result from raft command that should update a cluster-wide identity, but got invalid result structure for {id}. Leader is {LeaderTag}, Current node tag is {NodeTag}.");
+
+            var fullId = $"db/{databaseName}/Identities/{id}".ToLowerInvariant();
+            if (resultDict.ContainsKey(fullId) == false)
+            {
+                throw new InvalidOperationException(
+                    $"Expected to get result from raft command that should update a cluster-wide identity, but {id} was not in the result list. Leader is {LeaderTag}, Current node tag is {NodeTag}.");
+            }
+
+            var rc = resultDict[fullId];
+
+            if (rc == -1)
+            {
+                throw new InvalidOperationException(
+                    $"Expected to get result from raft command that should update a cluster-wide identity, but {id} was set but not able to be read. shouldn't reach here. Leader is {LeaderTag}, Current node tag is {NodeTag}.");
+            }
+
+            return resultDict[fullId];
```

Comment:
Why is this called multiple times?

Reply:
not sure I follow. Each throwing is a different case.  I will compact the code a little.